### PR TITLE
[Neo Wallet Fix] Add cert import exception

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

Dotnet has no implementation of `ExportParameters` for ECdsa, thus causing the `Import` method fail.

Fixes # (issue)

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Updated the Unit test of `TestImportCert` in UT_NEP6Wallet.cs

- [x] TestImportCert

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs [hunk 1]
```diff
@@ -223,6 +224,10 @@ public override IEnumerable<WalletAccount> GetAccounts()
 
         public override WalletAccount Import(X509Certificate2 cert)
```
Comment (roman-khimov) [inline src/Neo/Wallets/NEP6/NEP6Wallet.cs:225]: Are there any users of it?

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs [hunk 2]
```diff
@@ -223,6 +224,10 @@ public override IEnumerable<WalletAccount> GetAccounts()
...
         {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (roman-khimov) [inline src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: Can it be implemented with 3rd-party libraries for OSX?

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs (hunk 1 above)
Comment (shargon) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:225]: I don't think so

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs (hunk 2 above)
Comment (Jim8y) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: have tried, cant extract the private key. nothing in the ecdsa or cert is implementated.

**Review by shargon — APPROVED** [review]

**Comment by Jim8y** [comment]
time to fix the random error:

Warning: D:\a\neo\neo\src\Neo\SmartContract\ApplicationEngine.cs(256,27): warning CS0618: 'ApplicationEngine.GasConsumed' is obsolete: 'This property is deprecated. Use FeeConsumed instead.' [D:\a\neo\neo\src\Neo\Neo.csproj::TargetFramework=net8.0]
C:\Program Files\dotnet\sdk\8.0.300\Microsoft.Common.CurrentVersion.targets(2401,5): error MSB3248: Parameter "AssemblyFiles" has invalid value "D:\a\neo\neo\src\Neo\bin\Debug\net8.0\Neo.dll". The process cannot access the file 'D:\a\neo\neo\src\Neo\bin\Debug\net8.0\Neo.dll' because it is being used by another process. [D:\a\neo\neo\src\Plugins\RpcServer\RpcServer.csproj::TargetFramework=net8.0]
  Neo -> D:\a\neo\neo\src\Neo\bin\Debug\net8.0\Neo.dll
  Neo -> D:\a\neo\neo\src\Neo\bin\Debug\netstandard2.1\Neo.dll

**Review by cschuchardt88 — APPROVED** [review]

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs (hunk 2 above)
Comment (cschuchardt88) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: OpenSSL

Comment (Jim8y) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: not necessary, no one use this. exception is fine already.

Comment (cschuchardt88) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: If `openssl` is installed on the machine and configured right. `dotnet` will use it automatically.

**Comment by Jim8y** [comment]
> No reason to have this PR. Tests were working fine as before. So that shows that this works on macos.

What a disrespectful expression, feels like i am faking the issue:

```shell
Test method Neo.UnitTests.Wallets.NEP6.UT_NEP6Wallet.TestImportCert threw exception: 
Interop+AppleCrypto+AppleCFErrorCryptographicException: The operation couldn’t be completed. (OSStatus error -67738 - CSSM Exception: -2147415780 CSSMERR_CSP_INVALID_KEYATTR_MASK)
    at Interop.AppleCrypto.TrySecKeyCopyExternalRepresentation(SafeSecKeyRefHandle key, Byte[]& externalRepresentation)
   at System.Security.Cryptography.EccSecurityTransforms.TryExportDataKeyParameters(SecKeyPair keys, Boolean includePrivateParameters, ECParameters& ecParameters)
   at System.Security.Cryptography.EccSecurityTransforms.ExportParameters(Boolean includePrivateParameters, Int32 keySizeInBits)
   at Neo.Wallets.NEP6.NEP6Wallet.Import(X509Certificate2 cert) in /Users/jinghuiliao/git/neo/src/Neo/Wallets/NEP6/NEP6Wallet.cs:line 229
   at Neo.UnitTests.Wallets.NEP6.UT_NEP6Wallet.TestImportCert() in /Users/jinghuiliao/git/neo/tests/Neo.UnitTests/Wallets/NEP6/UT_NEP6Wallet.cs:line 305
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)

```

**Comment by Jim8y** [comment]
@neo-project/core finally fixed with BouncyCastle

**Review by shargon — COMMENTED** [review]
Use the same logic as:

https://github.com/neo-project/neo/blob/df067915f25f0632dcaf6ddc4325bb7ccef3034f/src/Neo/Wallets/Wallet.cs#L413-L423

**Comment by cschuchardt88** [comment]
> What a disrespectful expression, feels like i am faking the issue:

Is this on your computer? Then does the tests on `github`?

**Comment by shargon** [comment]
> Use the same logic as?
> 
> https://github.com/neo-project/neo/blob/df067915f25f0632dcaf6ddc4325bb7ccef3034f/src/Neo/Wallets/Wallet.cs#L413-L423

That's logic doesn't work in mac?

**Comment by cschuchardt88** [comment]
> That's logic doesn't work in mac?

It would have to be working. It is currently in the code right now. I think his computer isn't configured right.

**Comment by Jim8y** [comment]
> > That's logic doesn't work in mac?
> 
> It would have to be working. It is currently in the code right now. I think his computer isn't configured right.

@shargon please tell him how many code we have fixed that does not work on mac. And very funny, all of a sudden i need a `right` confuguration to run the neo test, even if i have being working on neo for many years.

**Review by cschuchardt88 — APPROVED** [review]

File: src/Neo/Wallets/NEP6/NEP6Wallet.cs (hunk 2 above)
Comment (shargon) [reply src/Neo/Wallets/NEP6/NEP6Wallet.cs:227]: but if it works with openssl installed, with this condition it wont never work

**Review by shargon — APPROVED** [review]

**Pull request merged**